File: javascript/isogram/isogram.js
const Isogram = function Isogram(word) {
  this.word = word;
};

/* Return true if an array has duplicates */
function hasDuplicates(array) {
  /* Convert to set (no duplicates) and compare sizes */
  return (new Set(array)).size !== array.length;
}

/* Create a regex that matches all non-letters, treating unicode characters
   correctly */
let xRegExp = require('xregexp');
/* Use global flag to match all */
const allNonlettersRegexp = xRegExp('\\PL', 'g');

/* Return true if the word is an isogram (no repeated letters) */
Isogram.prototype.isIsogram = function isIsogram() {
  /* Convert to lower case, remove all non-letter characters, and check to see
     if there are no duplicates */
  return !hasDuplicates(xRegExp.replace(this.word.toLowerCase(),
                                        allNonlettersRegexp, '').split(''));
};

module.exports = Isogram;

File: python/isogram/isogram.py
def is_isogram(s):
    sletters = list(filter(str.isalpha, s.lower()))
    return len(set(sletters)) == len(sletters)

def is_isogram_(s):
    seen = set()
    for c in s.lower():
        if not c.isalpha():
            continue
        if c in seen:
            return False
        seen.add(c)
    else:
        return True
